### PR TITLE
Add retention fields to all_subscriptions_v1

### PR DESCRIPTION
**Pull request by relud**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [x] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

### CODE REVIEW

File: bigquery_etl/format_sql/tokenizer.py [hunk 1]
```diff
@@ -410,7 +410,7 @@ class StatementSeparator(Token):
 class Operator(Token):
     """Operator."""
 
-    pattern = re.compile(r"<<|>>|>=|<=|<>|!=|.")
+    pattern = re.compile(r"<<|>>|>=|<=|=>|<>|!=|.")
```
Comment (relud) [inline bigquery_etl/format_sql/tokenizer.py:413]: bigquery supports named arguments via `=>` now, as seen in https://cloud.google.com/bigquery/docs/reference/standard-sql/interval_functions#make_interval

File: sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql [hunk 2]
```diff
@@ -20,7 +20,7 @@ WITH stripe_subscriptions AS (
     cancel_at,
     cancel_at_period_end,
     ended_at,
-    COALESCE(ended_at, CURRENT_TIMESTAMP) AS end_date,
+    COALESCE(ended_at, TIMESTAMP(CURRENT_DATE)) AS end_date,
```
Comment (relud) [inline sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql:23]: This was causing `months_retained` to potentially exceed `current_months_since_subscription_start`, which it should not do.

File: sql/mozfun/norm/diff_months/udf.sql [hunk 3]
```diff
@@ -1,32 +1,37 @@
-CREATE OR REPLACE FUNCTION norm.subscription_months_renewed(
+CREATE OR REPLACE FUNCTION norm.diff_months(
   start DATETIME,
   `end` DATETIME,
-  grace_days INT64
+  grace_period INTERVAL,
+  inclusive BOOLEAN
```
Comment (relud) [inline sql/mozfun/norm/diff_months/udf.sql:5]: In renamed this function to be more generic I added the `inclusive` argument to make it clear whether the interval from start to end is considered inclusive (jan 1 to feb 1 returns 1) or exclusive (jan 1 to feb 1 returns 0)

File: bigquery_etl/format_sql/tokenizer.py (hunk 1 above)
Comment (acmiyaguchi) [reply bigquery_etl/format_sql/tokenizer.py:413]: That's a really strange choice of syntax, although named arguments are a nice overall add.

File: sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql [hunk 4]
```diff
@@ -272,6 +276,18 @@ SELECT
     utm_medium,
     utm_source
   ).*,
+  mozfun.norm.diff_months(
+    start => DATETIME(subscription_start_date, plan_interval_timezone),
+    `end` => DATETIME(end_date, plan_interval_timezone),
+    grace_period => billing_grace_period,
+    inclusive => FALSE
+  ) AS months_retained,
```
Comment (acmiyaguchi) [inline sql/moz-fx-data-shared-prod/mozilla_vpn_derived/all_subscriptions_v1/query.sql:284]: So much easier to read with named arguments.

**Review by acmiyaguchi — APPROVED** [review]

**Pull request merged**